src/Aspire.Hosting.Azure/AzureStorageResource.cs

```diff
@@ -32,6 +32,14 @@ public class AzureStorageResource(string name) : Resource(name), IAzureResource
     /// </summary>
     public bool IsEmulator => this.IsContainer();
 
+    public string? GetConnectionString() => IsEmulator
+        ? AzureStorageEmulatorConnectionString.Create(
+            tablePort: GetEmulatorPort("table"),
+            queuePort: GetEmulatorPort("queue"),
+            blobPort: GetEmulatorPort("blob"))
+        // TODO: I'm not entirely sure what the behavior should be for when *not* emulated; the existing behavior component-specific URIs wouldn't be valid connection strings either.
+        : throw new DistributedApplicationException($"Azure storage resource does not generate non-emulated connection strings.");
```

Comment:
This is what I'm worried about. It means code in the app host need to handle when the emulator isn't in use as well.